File: hotel-server/src/main/java/com/hotel/entity/Room.java
package com.hotel.entity;

import com.hotel.constant.RoomStatus;
import com.hotel.constant.RoomType;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

/**
 * @author guangyong.yang
 * @date 2019-01-15
 * 房间
 */
@Getter
@Setter
@Entity
@Table(name = "room")
public class Room {
    /**
     * id
     */
    @Id
    @Column
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;


    /**
     * 房间号
     */
    @Column(length = 20,unique = true)
    private String roomNo;


    /**
     * 房间类型
     */
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private RoomType type;


    /**
     * 入住日期
     */
    @Column
    @Temporal(TemporalType.TIMESTAMP)
    private Date checkInDate;


    /**
     * 退房日期
     */
    @Column
    @Temporal(TemporalType.TIMESTAMP)
    private Date checkOutDate;


    /**
     * 客房状态
     */
    @Column
    @Enumerated(EnumType.STRING)
    private RoomStatus status;

    /**
     * 床
     */

    @OneToMany(cascade = CascadeType.ALL,mappedBy = "room")
    private List<Bed> beds;


    /**
     * 顾客
     */
    @ManyToMany(cascade = CascadeType.ALL)
    @JoinTable(name = "room_customer",inverseJoinColumns = @JoinColumn(name = "customer_id"),joinColumns = @JoinColumn(name = "room_id"))
    private List<Customer> customers;


    /**
     * 房价
     */
    @OneToMany(mappedBy = "room",cascade = CascadeType.ALL)
    private List<Charge> charges;

    /**
     * 收入记录
     */
    @OneToMany(mappedBy = "room",cascade = CascadeType.ALL)
    private List<Income> incomes;

}

File: hotel-front/src/app/dto/Room.ts
import {Bed} from './Bed';
import {Customer} from './Customer';
import {Income} from './Income';
import {Charge} from './Charge';

export class Room {
  id: number;

  /**
   * 房间号
   */
  roomNo: string;


  /**
   * 类型
   */
  type: string;
  /**
   * 床
   */
  beds: Bed[];

  /**
   * 客人
   */
  customers: Customer[];
  /**
   * 当前房价
   */
  money: number;


  /**
   * 入住日期
   */
  checkInDate: string;

  /**
   * 退房日期
   */
  checkOutDate: string;

  /**
   * 客房状态(0:空房;1:有客;2:不可订)
   */
  status: number;

  /**
   * 收入
   */
  incoming: Income[];

  /**
   * 定价记录
   */
  charges: Charge[];
}
